### PR TITLE
refactor: CSS file generation for Astra Dynamic CSS

### PR DESCRIPTION
**Pull request by Nikschavan**
Extends #1358

- Adds a separate `Astra_Filesystem` class which can be reused elsewhere where filesystem access is required.
- Separates cache logic to `Astra_Filesystem_Base` and `Astra_Filesystem` to make it easier to reuse logic from Astra Pro for it's similar caching.

### CODE REVIEW

File: inc/class-astra-cache.php [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+	 * Creates CSS files.
+	 *
+	 * @param  string $style_data   Gets the CSS for the current Page.
+	 * @param  string $timestamp    Gets the current timestamp.
+	 * @param  string $type         Gets the type theme/addon.
+	 * @param  string $assets_info  Gets the assets path info.
+	 * @since  x.x.x
+	 * @return void
+	 */
+	public function file_write( $style_data, $timestamp, $type, $assets_info ) {
```
Comment (Balachandark) [inline inc/class-astra-cache.php]: $type can now be removed

File: inc/class-astra-cache.php [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+	public function enqueue_styles( $style_data, $type ) {
+		if ( $this->inline_assets() ) {
+			wp_add_inline_style( 'astra-' . $type . '-css', $style_data );
+		} else {
+			$assets_info    = $this->get_asset_info( $style_data, $type );
+			$post_timestamp = $this->get_post_timestamp( $assets_info );
+
+			// Check if we need to create a new file or override the current file.
+			if ( ! empty( $style_data ) && true === $post_timestamp['create_new_file'] ) {
+				$this->file_write( $style_data, $post_timestamp['timestamp'], $type, $assets_info );
```
Comment (Balachandark) [inline inc/class-astra-cache.php]: $type should be removed

File: inc/class-astra-cache.php (hunk 1 above)
Comment (Nikschavan) [reply inc/class-astra-cache.php]: Done - removed the variable now.

File: inc/class-astra-cache.php (hunk 2 above)
Comment (Nikschavan) [reply inc/class-astra-cache.php]: Removed the variable now.

File: inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php [hunk 3]
```diff
@@ -23,9 +23,13 @@
...
 
-	$above_header_layout = astra_get_option( 'above-header-layout' );
+	$above_header_layout = astra_get_option( 'above-header-layout', 'disabled' );
 
-	if ( 'disabled' == $above_header_layout && false === Astra_Ext_Transparent_Header_Markup::is_transparent_header() ) {
+	if ( 'disabled' === $above_header_layout ) {
+		return $dynamic_css;
+	}
+
+	if ( false === Astra_Ext_Transparent_Header_Markup::is_transparent_header() ) {
```
Comment (UttamSharma007) [inline inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php]: Can be combined these 2 if condition with or operator?

File: inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php [hunk 4]
```diff
@@ -281,9 +285,13 @@ function astra_ext_transparent_above_header_sections_dynamic_css( $dynamic_css,
...
 	// set page width depending on site layout.
-	$below_header_layout = astra_get_option( 'below-header-layout' );
+	$below_header_layout = astra_get_option( 'below-header-layout', 'disabled' );
+
+	if ( 'disabled' === $below_header_layout ) {
+		return $dynamic_css;
+	}
 
-	if ( 'disabled' == $below_header_layout && false === Astra_Ext_Transparent_Header_Markup::is_transparent_header() ) {
+	if ( false === Astra_Ext_Transparent_Header_Markup::is_transparent_header() ) {
```
Comment (UttamSharma007) [inline inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php]: Can be combined these 2 if condition with or operator?

File: inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php (hunk 3 above)
Comment (Nikschavan) [reply inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php]: This was changed intentionally to fix one of the errors with transparent headers where the css for the transparent header was not being loaded.

File: inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php (hunk 4 above)
Comment (Nikschavan) [reply inc/addons/transparent-header/classes/dynamic-css/header-sections-dynamic.css.php]: This was changed intentionally to fix one of the errors with transparent headers where the css for the transparent header was not being loaded.

**Pull request merged**